packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,514 @@
+    newInstance.updateFeeRate();
+    return newInstance;
+  }
+
+  public costToAccelerate(): string {
+    return new BigNumber(this.requiredFee).minus(this.currentFee).toString();
+  }
+
+  public costToCancel(destinationAddress: string): string {
+    const cancelInstance = this.prepareCanceled(destinationAddress);
+    return new BigNumber(cancelInstance.requiredFee)
+      .minus(this.currentFee)
+      .toString();
+  }
+
+  public getAbsFees(customFeeRate?: FeeRateSatsPerVByte): string {
+    const feeRate = customFeeRate || this.targetFeeRate;
+    return new BigNumber(feeRate).multipliedBy(this.estimateVsize()).toString();
+  }
+
+  /**
+   * Updates sequence numbers for all inputs
+   *
+   * This method ensures that all inputs have a sequence number that signals RBF,
+   * as required by BIP125.
+   *
+   * @private
+   */
+  private updateSequenceNumbers(): void {
+    for (let i = 0; i < this.newPsbt.PSBT_GLOBAL_INPUT_COUNT; i++) {
+      const currentSequence = this.newPsbt.PSBT_IN_SEQUENCE[i];
+      if (currentSequence === null || currentSequence >= 0xfffffffe) {
+        this.newPsbt.PSBT_IN_SEQUENCE[i] = RBF_SEQUENCE;
+      }
+    }
+  }
+
+  /**
+   * Adjusts transaction outputs to accommodate the increased fee
+   *
+   * This method implements the logic for subtracting the fee increase from outputs,
+   * either from a specific output or proportionally from all non-change outputs.
+   *
+   * @private
+   * @throws {Error} If fee subtraction would create dust outputs
+   */
+  private adjustOutputs(): void {
+    if (this.feeOutputIndex !== undefined) {
+      this.subtractFeeFromSpecificOutput();
+    } else {
+      this.subtractFeeProportionally();
+    }
+  }
+
+  /**
+   * Subtracts the fee from a specific output
+   *
+   * This method reduces the amount of a specific output to cover the increased fee.
+   * If the resulting output would be dust, it throws an error.
+   *
+   * @private
+   * @throws {Error} If the specified output would become dust after fee subtraction
+   */
+  private subtractFeeFromSpecificOutput(): void {
+    const outputIndex = this.feeOutputIndex!;
+    const output = this.newPsbt.PSBT_OUT_AMOUNT[outputIndex];
+    if (output === undefined) {
+      throw new Error(`Output at index ${outputIndex} does not exist`);
+    }
+    const outputAmount = new BigNumber(output.toString());
+    const newAmount = BigNumber.max(
+      outputAmount.minus(this.feeIncrease),
+      new BigNumber(0),
+    );
+    if (newAmount.isLessThan(this.dustThreshold)) {
+      throw new Error(
+        `Output ${outputIndex} would become dust after fee subtraction`,
+      );
+    }
+    const newOutput = {
+      amount: Number(newAmount.toString()),
+      script: Buffer.from(this.newPsbt.PSBT_OUT_SCRIPT[outputIndex], "hex"),
+    };
+
+    this.newPsbt.deleteOutput(outputIndex);
+    this.newPsbt.addOutput(newOutput);
+  }
+
+  /**
+   * Subtracts the fee proportionally from all non-change outputs
```

Comment:
why non-change outputs? why not just all outputs?

Reply:
Well I did it because of these reasons :

1) Change Outputs = Reserved: Those are meant to return your leftover funds, so I did not want to touch them during fee adjustments , as the increase fees should come from the receiver side (or else we add additional UTXO to cover it ...)

2) Avoiding Dust: Subtracting from change could create dust outputs, which are impractical and can cause issues.

Reply:
This class is depreciated now